src/replica_ctrl.py

```diff
@@ -77,69 +86,33 @@ def get_tasks_ips(tasks, overlay_network_name):
     return tasks_ips
 
 
-def get_primary(tasks_ips, replicaset_name, mongo_port):
-    logger = logging.getLogger(__name__)
-    logger.info("searching primary")
-    for t in tasks_ips:
-        mc = pm.MongoClient(t, mongo_port)
-        try:
-            if mc.is_primary:
-                mc.close()
-                return t
-        except:
-            mc.close()
-            logger.info("{} is not primary".format(t))
-        mc.close()
-    return None
-
-def get_secondaries(tasks_ips, replicaset_name, mongo_port):
-    logger = logging.getLogger(__name__)
-    logger.info("searching secondaries")
-    for t in tasks_ips:
-        mc = pm.MongoClient(t, mongo_port)
-        try:
-            #it seams that mc.secondaries is not working
-            if mc.secondaries is not None and mc.secondaries != set():
-                logger.info("secondaries: {}".format(mc.secondaries))
-                mc.close()
-                return mc.secondaries
-        except:
-            mc.close()
-        mc.close()
-    return None
-
-
-def get_current_members(mongo_tasks_ips, replicaset_name, mongo_port):
-    current_ips = set()
-    logger = logging.getLogger(__name__)
-    for t in mongo_tasks_ips:
-        mc = pm.MongoClient(t, mongo_port)
-        try:
-            config = mc.admin.command("replSetGetConfig")['config']
-            for m in config['members']:
-                current_ips.add(m['host'].split(":")[0])
-            mc.close()
-        except:
-            mc.close()
-    logger.info("Current members in mongo configuration: {}".format(current_ips))
-    return current_ips
-
+def init_replica(mongo_tasks_ips, replicaset_name, mongo_port):
+    """
+    Init a MongoDB replicaset from the scratch.
 
-def create_cluster(mongo_tasks_ips, replicaset_name, mongo_port):
-    logger = logging.getLogger(__name__)
+    :param mongo_tasks_ips:
+    :param replicaset_name:
+    :param mongo_port:
+    :return:
+    """
     config = create_mongo_config(mongo_tasks_ips, replicaset_name, mongo_port)
+    logger = logging.getLogger(__name__)
     logger.info("Initial config: {}".format(config))
+
     # Choose a primary and configure replicaset
     primary_ip = list(mongo_tasks_ips)[0]
     primary = pm.MongoClient(primary_ip, mongo_port)
-    res = None
     try:
         res = primary.admin.command("replSetInitiate", config)
-    except:
-        res = primary.admin.command("replSetReconfig", config, force=True)
```

Comment:
Oh, I was thinking this method to handle only the "start from the scratch" subscenario (hence the temporary method renaming to init_replica). 

Mmm, OK, that's weird. If that's the case, let's make it come to that error again so that we can inspect the proper exception to be handled. Also, I'd like to better understand why we'd get empty list of ips in that case.